site/src/components/CodeExample/CodeExample.tsx

```diff
@@ -21,6 +22,9 @@ export const CodeExample: FC<CodeExampleProps> = ({
   // the secure option, not remember to opt in
   secret = true,
 }) => {
+  const [toShow, setToShow] = useState<boolean>(true)
+  secret = toShow;
```

Comment:
This strikes me as a mistake – render logic should not mutate component inputs. I haven't thought this through all the way, but my gut feeling is that you might want to use `secret` as the default value for your state, doing something like this:
```tsx
const [showSecret, setShowSecret] = useState(secret);
// Reference showSecret for all logic after this point
```
With this, the `secret` prop would inform the default value of the state, but could be overridden when necessary